### DISCUSSION TITLE
Laden pausieren

### DISCUSSION THREAD

**Discussion by phk0812** [goecharger/go-eCharger-API-v2#159] [Q&A]
Hallo,
es gibt einige API-keys, die vermuten lassen, dass man eine Ladung statt zu stoppen auch pausieren kann.
- modelStatus ... NotChargingBecauseMinPauseDuration
- acp ... allowChargePause
- mcpd ... minChargePauseDuration
- mcpea ... minChargePauseEndsAt

Leíder habe ich in der Doku nichts gefunden, wie man die Ladung über die API pausieren kann.
Bitte um Info, welchen API-key man dazu benutzen kann.
Danke

**Comment by 0xFEEDC0DE64** [comment]
forceState auf ForceOff stellen pausiert die Ladung

modelStatus ist sowieso readonly, es drückt aus warum gerade Ladeerlaubnis da ist oder warum nicht da ist (zb wegen der Auto-Kompatibilität MinPauseDuration, manche Autos spinnen wenn eine Ladeunterbrechung zu kurz ist, wir verlängern sie dann künstlich)
allowChargePause stellt auto-kompatibilität ein, ob das auto ladepausen überhaupt verträgt (hatten da auch schon spinnende autos)
minChargePauseDuration drückt aus, wie lange eine ladepause sein muss (= auto kompatibilität)
minChargePauseEndsAt drückt aus, wann die aktuelle min charge pause enden wird, und durch auf null setzen kann man es sofort abbrechen und losladen

**Reply by phk0812** [reply]
Ok, dass mit dem frc wusste ich.
Das ist aber nach meinem Verständnis keine Pause, sondern ein Stoppen.
Sonst würde bei einem frc=1 der ModelStatus ja auf NotChargingBecauseMinPauseDuration wechseln statt auf NotChargingBecauseForceStateOff.

Die Frage ist nun:
Welchen ApiKey muss ich setzen, damit der ModelStatus auf NotChargingBecauseMinPauseDuration wechselt.

Die beiden minChargePause-Keys machen für mich den Eindrück, dass man die Ladung unterbrechen könne und diese dann gem. der in diesen Keys eingestellten Zeit automatisch wieder startet.

Zum Verständnis, weswegen ich das suche:
Beim PV-Überschussladen, muss man zeitweise die Ladung pausieren.
Bei frc=1 denkt das Auto aber, dass die Ladung beendet wurde und schickt mir jedesmal eine Benachrichtigung.
Es wäre also super, wenn man die Ladung so einstellen könnte, dass das Auto zwar noch offiziell lädt, aber weiß, dass es derzeit keinen Strom beziehen darf.
Also quasi eine Art amp=0.

**Reply by martiko70** [reply]
Genau das würde ich auch brauchen... das Thema ist ja schon recht alt, gibt es dazu inzwischen eine Lösung oder eine Idee?

**Reply by 0xFEEDC0DE64** [reply]
wenn man einstellt, dass eine (automatisch bedingte) Ladepause nie länger als 29 minuten sein darf (weil das auto nach 30 minuten endgültig schlafen geht und dann nie wieder aufwachen würde), dann starten wir immer wieder kurz die ladung für 1 sekunde.

Manche autos haben auch gesponnen, wenn eine automatische Ladepause zu kurz war, die haben dann einen Versorgungsfehler angenommen. Der go-eCharger verlängert dann künstlich die Ladepause, obwohl scheduler oder OCPP oder last management schon lange das Laden wieder erlauben.

man stellt also die minimale Ladepause auf 10min, dreht den forceState für 2min auf force-Off, danach wieder auf neutral, der charger sollte damit nicht nur 2min Ladepause machen, sondern eben 10min (= die minimale Ladepause).

Wo liegt der Untschied zwischen einer pausierten und gestoppten Ladung?

In der Firmware wird ein Ladestopp durch abstecken des Autos erkannt, damit wird dann zb die Transaktion (RFID oder OCPP) beendet, der Ladesitzungs-Energiezähler wird auf 0 gestellt usw.

forceState=off ist eine gleiche lade-pause wie "ocpp verbietet das laden", "temperatur verbietet das laden" oder laden nicht erlaubt weil authentifizierung erforderlich ist.
